medical_compliance/medical_compliance/api/tasks.py

```diff
@@ -12,58 +12,35 @@
+from models import WeightMeasurement
+from analyzers.weight_analyzers import analyze_weights
 
-logger = get_task_logger("medical_compliance.fetch_measurement")
+logger = get_task_logger("medical_compliance_measurements.fetch_weight_measurement")
 
 
 app = Celery('api.tasks', broker=settings.BROKER_URL)
 app.conf.update(
-    CELERY_DEFAULT_QUEUE='withings_measurements',
+    CELERY_DEFAULT_QUEUE='medical_compliance_measurements',
     CELERY_QUEUES=(
-        Queue('withings_measurements', Exchange('withings_measurements'), routing_key='withings_measurements'),
+        Queue('medical_compliance_measurements', Exchange('medical_compliance_measurements'), routing_key='medical_compliance_measurements'),
     ),
 )
 
 
-@app.task(name='medical_compliance.fetch_measurement')
-def fetch_measurement(userid, start_ts, end_ts, measurement_type_id):
+@app.task(name='medical_compliance_measurements.fetch_weight_measurement')
+def fetch_weight_measurement(user_id, input_source, measurement_unit, timestamp, timezone, value):
     logger.debug(
-        "Sending request for measurement retrieval for userid: %s, start ts: %s, end ts: %s, type: %s",
-        str(userid),
-        str(start_ts),
-        str(end_ts),
-        str(measurement_type_id)
+        "Received weight measurement for user_id: %s, input_source: %s",
+        str(user_id),
+        str(input_source)
     )
 
-    credentials = WithingsCredentials(access_token=settings.WITHINGS_OAUTH_V1_TOKEN,
-                                      access_token_secret=settings.WITHINGS_OAUTH_V1_TOKEN_SECRET,
-                                      consumer_key=settings.WITHINGS_CONSUMER_KEY,
-                                      consumer_secret=settings.WITHINGS_CONSUMER_SECRET,
-                                      user_id=userid)
-    # TODO: modify storage of user credentials in settings file to a per userid basis
-
-    client = WithingsApi(credentials)
-    response = client.request('measure', 'getmeas', {
-        'startdate': start_ts, 
-        'enddate': end_ts,
-        'meastype': int(measurement_type_id)
-    })
-    measures = WithingsMeasures(response)
-    timezoneStr = response['timezone']
-    measurement_type = WithingsMeasurement.get_measure_type_by_id(int(measurement_type_id))
-
-    for m in measures:
-        meas = WithingsMeasurement(
-            withings_user_id = int(userid),
-            type=measurement_type_id,
-            retrieval_type=m.attrib,
-            measurement_unit=WithingsMeasurement.MEASUREMENT_SI_UNIT[measurement_type],
-            timestamp=m.data['date'],
-            timezone=timezoneStr,
-            value=m.__getattribute__(measurement_type))
-        meas.save()
-
-
-
+    weightMeas = WeightMeasurement(
+            user_id = int(user_id),
+            input_source=input_source,
+            measurement_unit=measurement_unit,
+            timestamp=timestamp,
+            timezone=timezone,
+            value=value)
+    weightMeas.save()
```

Comment:
Try to be consistent with naming conventions: either use camel case or underscores but not both. I think the standard for python is underscores. Also avoid abbreviations.